### ISSUE TITLE
Slow local storage upload because of safe filename check

### ISSUE THREAD

**Issue by denolfe** [payloadcms/payload#7192]
### Describe the Bug

Storing a single image in Payload takes ~1-2 seconds at a minimum because of the safe filename check that is performed in order to determine if the filename is already taken. If it is, we append `-<number>` on the end. 

It was reported that explicitly setting the filename to be unique didn't help, either.

One possible solution is to append a unique string on all uploads. This would avoid having to do the uniqueness check. Ideally, this would be behind a flag if implemented.

### To Reproduce

.

### Payload Version

2.x

**Comment by tak-amboss** [comment]
> One possible solution is to append a unique string on all uploads. This would avoid having to do the uniqueness check. Ideally, this would be behind a flag if implemented.

Instead of appending a unique string on all uploads, you could also append a unique string instead of the `-<number>`. This would still require the filename check, but only once, which is certainly acceptable. This way it wouldn't be a breaking change and quite straight forward.

**Comment by JarrodMFlesch** [comment]
> Instead of appending a unique string on all uploads, you could also append a unique string instead of the -<number>. This would still require the filename check, but only once, which is certainly acceptable. This way it wouldn't be a breaking change and quite straight forward.

Could perform a count query with the right filtering/regex? Even a unique string has the _chance_ that it will not be unique.

**Comment by r1tsuu** [comment]
For this, I created a `beforeOperation` hook does append UUID to the filename
```ts
  ({ args, operation, req }) => {
        if ((operation === 'update' || operation === 'create') && req.file && req.file.name) {
          const parsed = path.parse(req.file.name);

          const newName = `${parsed.name}-${crypto.randomUUID()}${parsed.ext}`;

          if (args.data) {
            args.data.originalFilename = req.file.name;
            args.data.filename = newName;
          }

          req.file.name = newName;
        }
      }
```

**Comment by PatrikKozak** [comment]
Hey @denolfe - could you check if this issue is still reproducible in v3?

If so, could you please update this thread or open a new issue referencing this one.

We’ll be closing out older v2 issues soon as we transition our focus solely to v3 and future releases. Thanks!

**Comment by PatrikKozak** [comment]
@denolfe Closing this now per my comment above. Please feel free to open a new issue referencing this one if you are able to reproduce this in v3.
